basiclibrary/src/test/java/basiclibrary/LibraryTest.java

```diff
@@ -6,9 +6,76 @@
 import org.junit.Test;
 import static org.junit.Assert.*;
 
-public class LibraryTest {
+public class  LibraryTest {
+    //Feature 1
     @Test public void testSomeLibraryMethod() {
         Library classUnderTest = new Library();
-        assertTrue("someLibraryMethod should return 'true'", classUnderTest.someLibraryMethod());
+        assertEquals("The new array should have the same number of length as the input number",3, classUnderTest.roll(3).length);
+        for(int n: classUnderTest.roll(3)){
+            assertTrue("Each number should be greater or equal to 1",n>=1);
+            assertTrue("Each number should be less or equal to 6",n<=6);
+        }
     }
+
+   //Feature 2
+    @Test
+    public void checkarrayhasDuplicates(){
```

Comment:
You want to make sure you're method names are in proper camelCase!

Reply:
Thanks!